### ISSUE TITLE
Compiler error when a class has the same name as it's containing namespace

### ISSUE THREAD

**Issue by Porl91** [bridgedotnet/Bridge#3970]
Apologies for the slightly convoluted example. The bare bones of the issue are:

- There's a class named `Configuration` that's defined within a namespace of the same name.
- There's a `SpecialList` class that has an `Update` method that takes an `update` lambda. In this example I have an `Update` call inside an `Update` call.

When these two areas of the code are combined it produces the following compile-time error:
`Bridge.Contract.EmitterException: Language construction NamedArgumentExpression is not supported`

Even though the error relates to named arguments, removing the `updater` named argument doesn't eliminate the error.

If I either rename the `Configuration` class to another than that of it's containing namespace, or I remove the nested `Update` call, the compiler error vanishes.

I can work around this issue by just renaming the class, but I thought I'd report it.

Here's a repro on deck:
https://deck.net/dc7122413f02c6aca5c0a17b75df7bd5

If you need any other info then just give me a shout - Cheers

**Comment by fabriciomurta** [comment]
Hello, @Porl91 ! You seem to have just hit something we couldn't implement yet:

Here's the code block it probably hit: [Compiler/Translator/Emitter/TempEmitter.cs:1540-1543](https://github.com/bridgedotnet/Bridge/blob/b027e975cb01df201c4f9ba03a2a28426ecffec7/Compiler/Translator/Emitter/TempEmitter.cs#L1540-L1543).

Now we'll have it logged as a feature and it would help us and others know this is a wanted part to get implemented on Bridge!

**Comment by Porl91** [comment]
Hi @fabriciomurta, thanks for the reply. I should have mentioned this in the OP, but this code has existed for a while, but the compiler error only started to occur when upgrading from Bridge **17.1.1** to **17.7.0**. Was this feature removed in between these versions?

**Comment by geoffreymcgill** [comment]
A fix for this issue has been created. Once a review and final testing is complete, the fix will be included in the upcoming Bridge 17.8.0 release.

**Comment by fabriciomurta** [comment]
Hello @Porl91, as you pointed out, it was really something that changed and made the code interpret as if it needed an unsupported feature, which was not the case. I also noticed that if you just remove the named parameter `updater: ` from the inner call to `.Update()`, is -also- works.

Anyway, the fix submitted by @vladsch looks great -- and works great!.. And this case won't go unnoticed again if it dare happen again, for we added the deck you provided as a unit test to be run every update.

As said by geoffrey, this fix is going to be included in the next Bridge release!
